magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleLimiter.kt

```diff
@@ -4,9 +4,9 @@ import android.content.Context
 import com.wealthfront.magellan.OpenForMocking
 
 @OpenForMocking
-public class LifecycleLimiter : LifecycleOwner, LifecycleAware {
-
-  private val lifecycleRegistry = LifecycleRegistry()
+public class LifecycleLimiter(
+  private val lifecycleRegistry: LifecycleRegistry
+) : LifecycleOwner, LifecycleAware {
```

Comment:
`LifecycleRegistry`s are pretty cheap right? I'm not sure I like the pattern of passing `LifecycleRegistry`s around, it kind of breaks the encapsulation that we otherwise have.

One subtle bug that could show up as a result is the order in which children receive events. If another object (like LifecycleLimiter here) can muck around in registry, it can add things in an unpredictable order. Since we dispatch lifecycle events to children in the order they're attached to the lifecycle, this can cause unpredictable behavior.

Since `LifecycleRegistry`s are pretty cheap, we should consider encouraging people to use those rather than break encapsulation IMO.